python/composio/tools/toolset.py

```diff
@@ -1157,7 +1170,16 @@ def get_expected_params_for_user(
         """
         # If `integration_id` is provided, use it to fetch the params
         if integration_id is not None:
-            return self._get_expected_params_from_integration_id(id=integration_id)
+            response = self._get_expected_params_from_integration_id(id=integration_id)
+            if auth_scheme is not None and response["auth_scheme"] != auth_scheme:
+                raise ComposioSDKError(
```

Comment:
Good addition of the auth scheme validation. This ensures that the provided auth scheme matches the one associated with the integration ID, preventing potential mismatches. Consider adding a comment explaining the rationale behind this check for future maintainers.